oauthenticator/github.py

```diff
@@ -141,10 +208,23 @@ def _check_organization_whitelist(self, org, username, access_token):
                 return True
         return False
 
+    @gen.coroutine
+    def _get_user_organizations(self, access_token):
+        http_client = AsyncHTTPClient()
+        headers = _api_headers(access_token)
+        next_page = "https://%s/user/orgs" % (GITHUB_API)
+        orgmap = {}
+        while next_page:
+            req = HTTPRequest(next_page, method="GET", headers=headers)
+            resp = yield http_client.fetch(req)
+            resp_json = json.loads(resp.body.decode('utf8', 'replace'))
+            next_page = next_page_from_links(resp)
+            for entry in resp_json:
+                orgmap[entry["login"]] = entry["id"]
```

Comment:
1) (unless you go with my original proposal of feature switches, but that does get slightly gross with respect to choosing them/adding environment variables) is definitely true.  (So, for that matter, does getting the GitHub email unless the user has a public primary email.)

I hadn't really thought about using the info at spawn time rather than in the authenticator class.  I guess that might work.  I don't know that I am likely to get this buttoned down before JupyterCon, though.  Today and tomorrow are full of job interviews, and then I need to stand up our JupyterLab tutorial environment for our All-Hands meeting next week...and then JupyterCon is upon us.